### PR TITLE
A0-2778: Bound notification block sync channel

### PR DESCRIPTION
**Pull request by ggawryal**
# Description

Replaced two channels in gossip network (`mpsc_notification_stream_authentication` and `mpsc_notification_stream_block_sync`) from tracing_unbounded to bounded channels, discarding new messages when buffer limit was hit and logging warning. For convenience, I replaced both these channels instead of just `mpsc_notification_stream_block_sync`.

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

# Checklist:

### CODE REVIEW

File: finality-aleph/src/network/gossip/service.rs [hunk 1]
```diff
@@ -5,14 +5,16 @@ use std::{
...
+    StreamExt,
+};
 use log::{debug, error, info, trace, warn};
 use network_clique::SpawnHandleT;
 use rand::{seq::IteratorRandom, thread_rng};
-use sc_utils::mpsc::{tracing_unbounded, TracingUnboundedReceiver, TracingUnboundedSender};
 use tokio::time;
 
-const QUEUE_SIZE_WARNING: usize = 1_000;
+const MAX_QUEUE_SIZE: usize = 1_000;
```
Comment (woocash2) [inline finality-aleph/src/network/gossip/service.rs:17]: We've recently seen the limit of 1000 being hit - just a question, if we're gonna drop the messages, do we want to increase this limit?

File: finality-aleph/src/network/gossip/service.rs [hunk 2]
```diff
@@ -104,6 +106,34 @@ impl<D: Data, P: Clone + Debug + Eq + Hash + Send + 'static> Network<D> for Serv
...
+struct NamedFuturesBoundedSender<T> {
+    name: &'static str,
+    tx: mpsc::Sender<T>,
+}
+
+impl<T> NamedFuturesBoundedSender<T> {
+    pub fn send(&mut self, msg: T) -> Result<(), TrySendError<T>> {
+        self.tx.try_send(msg).map_err(|e| {
+            if e.is_full() {
+                warn!(
```
Comment (timorl) [inline finality-aleph/src/network/gossip/service.rs:118]: Logging should rather happen outside, next to the other `if` you changed. In particular then you cannot (but also don't have to) use the name – from context you will know which one it is, and even better, to whom the dropped message was addressed. Since you don't have to do that you can just use a plain `Sender`, so you don't even need this struct, just construct a `mpsc::channel` and you are done.

tl;dr You put in too much effort, you need more laziness in your life. ;p

**Review by timorl — CHANGES_REQUESTED** [review]

File: finality-aleph/src/network/gossip/service.rs (hunk 1 above)
Comment (timorl) [reply finality-aleph/src/network/gossip/service.rs:17]: If anything probably decrease – neither block sync nor authentications should be sending that many messages to a single node ever.

Comment (ggawryal) [reply finality-aleph/src/network/gossip/service.rs:14]: I'd keep it as it is and let us make relevant benchmarks to determine optimal value there later, in the task related to bounding memory usage in the whole aleph-node - probably this value should be editable in some config anyway.

File: finality-aleph/src/network/gossip/service.rs (hunk 2 above)
Comment (ggawryal) [reply finality-aleph/src/network/gossip/service.rs]: Good point, tried to abstract the underlying channel, but I agree it is a bit unnecessary.

File: finality-aleph/src/network/gossip/service.rs [hunk 3]
```diff
@@ -200,11 +193,18 @@ impl<N: RawNetwork, AD: Data, BSD: Data> Service<N, AD, BSD> {
     fn send_to_authentication_peer(&mut self, data: AD, peer: N::PeerId) -> Result<(), SendError> {
         match self.get_authentication_sender(&peer) {
             Some(sender) => {
-                match sender.unbounded_send(data) {
+                match sender.try_send(data) {
                     Err(e) => {
                         // Receiver can also be dropped when thread cannot send to peer. In case receiver is dropped this entry will be removed by Event::NotificationStreamClosed
```
Comment (woocash2) [inline finality-aleph/src/network/gossip/service.rs:198]: is this comment relevant to `e.is_full()`?

File: finality-aleph/src/network/gossip/service.rs [hunk 4]
```diff
@@ -200,11 +193,18 @@ impl<N: RawNetwork, AD: Data, BSD: Data> Service<N, AD, BSD> {
...
                         // No need to remove the entry here
-                        if e.is_closed() {
+                        if e.is_full() {
+                            warn!(
+                                target: "aleph-network",
+                                "Failed sending data in mpsc_notification_stream_authentication to peer because peer_sender receiver is full: {:?}",
```
Comment (woocash2) [inline finality-aleph/src/network/gossip/service.rs:203]: Do we wanna use these names `mpsc_notification_stream_authentication` if actually nothing's named like this anymore? We could have: "Failed sending data through authentication notification channel ..." or sth like that.

**Review by woocash2 — APPROVED** [review]
Added some small comments, other than that looks great

File: finality-aleph/src/network/gossip/service.rs (hunk 4 above)
Comment (timorl) [reply finality-aleph/src/network/gossip/service.rs:203]: Yeah, no point keeping the weird name, the alternative you suggest is better.

File: finality-aleph/src/network/gossip/service.rs (hunk 3 above)
Comment (timorl) [reply finality-aleph/src/network/gossip/service.rs:198]: Not really, it should be only before `is_disconnected`.

**Review by timorl — APPROVED** [review]
Looks good, assuming @woocash2 's comments will get resolved.

File: finality-aleph/src/network/gossip/service.rs (hunk 3 above)
Comment (ggawryal) [reply finality-aleph/src/network/gossip/service.rs]: Good catch

File: finality-aleph/src/network/gossip/service.rs (hunk 4 above)
Comment (ggawryal) [reply finality-aleph/src/network/gossip/service.rs]: Agreed & renamed
